### ISSUE TITLE
[FEATURE] Adding variables to Event Listeners requires the variable to have a Changed Event

### ISSUE THREAD

**Issue by mutmedia** [AdamRamberg/unity-atoms#173]
**Describe the bug**
Adding variables to Event Listeners requires the variable to have a Changed Event, which, removes the ease of being able to directly add a variable to a listener

**To Reproduce**
1. Create an Event Listener of any type.
2. Change its Event Reference to be a Variable
3. Create a new variable with the "create" button.
4. Make some logic to change the Variable's value at runtime.
5. Hit play and notice the following error message: 
```
NotSupportedException: Event type UnityAtoms.BaseAtoms.FloatEvent not supported! Use UnityAtoms.BaseAtoms.FloatEvent or UnityAtoms.BaseAtoms.FloatPairEvent.
UnityAtoms.AtomVariable`5[T,P,E1,E2,F].GetEvent[E] () (at C:/Users/gusta/Desktop/gamedev/_unity packages/unity-atoms/Packages/Core/Runtime/Variables/AtomVariable.cs:247)
UnityAtoms.AtomEventReference`5[T,V,E,VI,EI].get_Event () (at C:/Users/gusta/Desktop/gamedev/_unity packages/unity-atoms/Packages/Core/Runtime/EventReferences/AtomEventReference.cs:30)
UnityAtoms.AtomBaseEventReference`3[T,E,EI].GetEvent[EO] () (at C:/Users/gusta/Desktop/gamedev/_unity packages/unity-atoms/Packages/Core/Runtime/EventReferences/AtomBaseEventReference.cs:100)
UnityAtoms.AtomEventReferenceListener`4[T,E,ER,UER].get_Event () (at C:/Users/gusta/Desktop/gamedev/_unity packages/unity-atoms/Packages/Core/Runtime/Listeners/AtomEventReferenceListener.cs:23)
UnityAtoms.AtomBaseListener`3[T,E,UER].OnEnable () (at C:/Users/gusta/Desktop/gamedev/_unity packages/unity-atoms/Packages/Core/Runtime/Listeners/AtomBaseListener.cs:56)
```
6. The error goes away if the variable has a Changed event

**Expected behavior**
When creating a new Variable for an Event Listener as described above, it shouldn't be required that the Variable has a Changed event, as it defeat the ease of being able to create the Variable there in the first place. It would be interesting if the variable itself could be treated as its own changed event, but for ease of implementation, the Changed event could be lazily created on demand.

**Screenshots**
![image](https://user-images.githubusercontent.com/4412564/85972492-0726e580-b985-11ea-93de-10f4ffde31b9.png)
![image](https://user-images.githubusercontent.com/4412564/85972508-0ee68a00-b985-11ea-9481-b9324bf4a2a5.png)
![image](https://user-images.githubusercontent.com/4412564/85972597-46553680-b985-11ea-8e94-49e799988e86.png)

**Environment**
 - OS: Windows
 - Unity Version 2019.2.19
 - Version of Unity Atoms 4.2.1

**Comment by AdamRamberg** [comment]
Hi @mutmedia! After some thought I think that I agree  reagarding the above and it would be nice if Events on Variables actually are instantiated if null already in their property getter. It would probably make the library easier to use. @soraphis Would like your opinion on this as well. 

My proposed solution would be to convert the events (`Changed` and `ChangedWithHistory)` to properties:
```
        /// <summary>
        /// Changed Event triggered when the Variable value gets changed.
        /// </summary>
        /// <value>Get the Changed Event.</value>
        public E1 Changed
        {
            get
            {
                if (_changed == null)
                {
                    _changed = ScriptableObject.CreateInstance<E1>();
                    _changed.name = $"CreatedAtRuntime_{typeof(T).Name}Event";
                }
                return _changed;
            }
            set
            {
                _changed = value;
            }
        }

        /// <summary>
        /// Changed Event triggered when the Variable value gets changed.
        /// </summary>
        [SerializeField]
        [FormerlySerializedAs("Changed")]
        private E1 _changed;
```
Corresponding changed would also need to be done for `ChangedWithHistory`.

The above change is pretty straight forward. However, if just do the above change you will notice that no Event is actually being attached to the Variable. That is because of how the `AtomVariableEditor` works. Did some initial tries locally to get that working and it was harder then it looked at the surface. I might give it another shoot later, but would also appreciate if someone would contribute and make a PR for this.

**Comment by AdamRamberg** [comment]
I will relabel this one as an enhancement, since what @mutmedia is describing is actually the initial intended behaviour, even though it might be a good one.

**Comment by mutmedia** [comment]
Had it basically implemented as you suggested already, sent out a PR!

**Comment by mutmedia** [comment]
Fixed in #179

**Comment by AdamRamberg** [comment]
Fixed in #179
